### PR TITLE
[Communication]: Added exception to begin_update_phone_number_capabilities

### PR DESCRIPTION
**Pull request by lsundaralingam**
This is to address the case when the service returns a response with the status code 200 and an error in the body.

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 1]
```diff
@@ -189,13 +190,19 @@ def begin_update_phone_number_capabilities(
...
-        return self._phone_number_client.phone_numbers.begin_update_capabilities(
+        poller = self._phone_number_client.phone_numbers.begin_update_capabilities(
             phone_number,
             calling=calling,
             sms=sms,
             **kwargs
         )
 
+        result_properties = poller.result().additional_properties
+        if 'status' in result_properties and result_properties['status'] == 'failed':
```
Comment (minnieliu) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:201]: Is this check case sensitive?

**Review by jbeauregardb — APPROVED** [review]
Thank you for doing this!

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py (hunk 1 above)
Comment (lsundaralingam) [reply sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:201]: Yes, this check is case sensitive.

Comment (lsundaralingam) [reply sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py]: Discussed this offline. The status value isn't always lower case, so this check should handle that.

**Review by beltr0n — APPROVED** [review]

**Pull request merged**